src/ParseServer.js

```diff
@@ -98,17 +110,6 @@ class ParseServer {
         }
       });
 
-    if (cloud) {
-      addParseCloud();
-      if (typeof cloud === 'function') {
-        cloud(Parse);
-      } else if (typeof cloud === 'string') {
-        require(path.resolve(process.cwd(), cloud));
-      } else {
-        throw "argument 'cloud' must either be a string or a function";
-      }
-    }
-
     if (security && security.enableCheck && security.enableCheckLog) {
```

Comment:
Looking at this change, I think we can easily move this to into a `.then(async () => {` below the server start complete. I would consider this part of the refactoring.

Reply:
That was what my previous [comments ](https://github.com/parse-community/parse-server/pull/7525#issuecomment-912561497) were about. The issue is that it will break tests unless we also implement a way to ensure init is completed before returning.

Ex: 
```js
const server = await new ParseServer(config);
await server.isReady()
```

Reply:
Oh I see, which tests will it break?

Reply:
Correct me if i'm wrong, but constructors can't be async, so the await will have to be in `isReady` function only and not the `ParseServer` constructor.